online-shop/src/app/admin.guard.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { AuthService } from './services/auth.service';
+import { Router } from '@angular/router';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class AdminGuard implements CanActivate {
+    constructor(private authService: AuthService, private router: Router) {}
+
+    canActivate(
+        route: ActivatedRouteSnapshot
+    ): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
```

Comment:
in this case I think you can keep only the boolean type, the 'Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree' is just an example of what types it can return, since you just return true/false, you can leave only boolean